core/src/main/scala/org/broadinstitute/workbench/ccm/model.scala

```diff
@@ -1,17 +1,325 @@
+  def asString: String
+}
+
+object Status {
+  final val RUNNING = "Running"
+  final val NOTSTARTED = "NotStarted"
+  final val STARTING = "Starting"
+  final val FAILED = "Failed"
+  final val DONE = "Done"
+
+  final val allStatuses = Seq(Running, NotStarted, Starting, Failed, Done)
+  final val inFlightStatuses = Seq(Running, NotStarted, Starting)
+  final val terminalStatuses = Seq(Failed, Done)
+
+  val stringToStatus = Map(
+    RUNNING -> Running,
+    NOTSTARTED -> NotStarted,
+    STARTING -> Starting,
+    FAILED -> Failed,
+    DONE -> Done
+  )
+
+  case object Done extends Status {
+    def asString = DONE
+  }
+
+  case object Running extends Status {
+    def asString = RUNNING
+  }
+
+  case object NotStarted extends Status {
+     def asString = NOTSTARTED
+  }
+
+  case object Starting extends Status {
+    def asString = STARTING
+  }
+
+  case object Failed extends Status {
+    def asString = FAILED
+  }
+  //see if there are others?
+
+}
+
+sealed trait BackEnd {
+  def asString: String
+}
+
+object BackEnd {
+  final val JES = "JES"
+  final val PAPIV2 = "PAPIV2" //???
+
+  final val allBackEnds = Seq(Jes, PapiV2)
+
+  val stringToBackEnd = Map(
+    JES -> Jes,
+    PAPIV2 -> PapiV2
+  )
+
+  case object Jes extends BackEnd {
+    def asString = JES
+  }
+
+  case object PapiV2 extends BackEnd {
+    def asString = PAPIV2
+  }
+
+}
+
 final case class CpuNumber(asInt: Int) extends AnyVal
 final case class BootDiskSizeGb(asInt: Int) extends AnyVal
-final case class Ram(asString: String) extends AnyVal
 final case class WorkflowId(id: String) extends AnyVal
-final case class DiskName(asString: String) extends AnyVal
+final case class DiskName(asString: String) extends AnyVal  //
 final case class DiskSize(asInt: Int) extends AnyVal
-final case class DiskType(asString: String) extends AnyVal
-final case class Preemptible(asInt: Int) extends AnyVal
-final case class MetadataResponse(value: List[Call]) extends AnyVal
+final case class PreemptibleAttemptsAllowed(asInt: Int) extends AnyVal
+final case class Attempt(asInt: Int) extends AnyVal
+final case class ExecutionEventDescription(asString: String) extends AnyVal
+
+
+final case class ExecutionEvent(description: ExecutionEventDescription,
+                                startTime: Instant,
+                                endTime: Instant)
+
+final case class Call(runtimeAttributes: RuntimeAttributes,
+                      executionEvents: List[ExecutionEvent],
+                      isCallCaching: Boolean,
+                      preemptible: Boolean,
+                      region: Region,
+                      status: Status,
+                      machineType: MachineType,
+                      backend: BackEnd,
+                      attempt: Attempt)
+
+final case class MetadataResponse(calls: List[Call], startTime: Instant, endTime: Instant)
+
+
+final case class RuntimeAttributes(cpuNumber: CpuNumber,
+                                   disks: Disks,
```

Comment:
can there be more than one disk?

Reply:
I'm actually not sure, I will ask Cromwell people

Reply:
The answer is, I'll turn it into a list